common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala

```diff
@@ -3898,6 +3899,14 @@ object CelebornConf extends Logging {
       .intConf
       .createWithDefault(3)
 
+  val WORKER_WRITER_GATHER_API_ENABLED: ConfigEntry[Boolean] =
+    buildConf("celeborn.worker.writer.gatherApi.enabled")
+      .categories("worker")
+      .version("0.6.0")
+      .doc("Worker will use gather API if this config is true.")
+      .booleanConf
+      .createWithDefault(false)
```

Comment:
it might be worth enabling by default given it's JDK feature, which generally is much stable.